src/renderer/PresetsBar.tsx

```diff
@@ -114,28 +116,41 @@ const PresetsBar = () => {
   ]);
 
   const handleCreatePreset = (prev: string[]) => {
-    if (prev.indexOf(presetName) !== -1) {
-      // TODO: overwrite preset values for the existing preset
-    } else {
-      // TODO: add code for creating a preset
+    savePreset(presetName);
+
+    // If we are creating a new preset and not just updating an existing one
+    if (prev.indexOf(presetName) === -1) {
       setSelectedPresetName(presetName);
       const newPresets = [...prev, presetName].sort();
       setPresetNames(newPresets);
     }
+
+    console.log('Finished Handling Create Preset!');
   };
 
   const handleLoadPreset = () => {
-    // TODO: add code for loading a preset
+    if (selectedPresetName) {
+      loadPreset(selectedPresetName)
+        .then((json_string) => {
+          performHealthCheck();
+          const jsonContent = JSON.parse(json_string);
+          console.log('result');
+          console.log(jsonContent);
+          return jsonContent;
+        })
+        .catch((error) => {
+          console.log(`failed to get preset ${selectedPresetName}`);
+          console.log(error);
+        });
+    }
+    console.log('done loading preset!');
   };
 
   const handleChangeNewPresetName = (newValue: string) => {
     setPresetName(newValue);
 
-    // Update selectedPresetName if the input value matches an existing preset name
-    const newSelectedName = presetNames.some((n) => n === newValue)
-      ? newValue
-      : undefined;
-    setSelectedPresetName(newSelectedName);
+    // TODO: If the preset name is an existing one, we should set the selected one to be that so we can load it
+    setSelectedPresetName(undefined);
```

Comment:
Oops, another case of my changes being reverted by accident. Could you undo this change?